File: FizzBuzz.go
package main
import "fmt"
func main() {
	for n := 1; n < 101; n++ {
		if CheckDivisibility(n, 15) {
			fmt.Println("FizzBuzz")
		} else if CheckDivisibility(n, 3) {
			fmt.Println("Fizz")
		} else if CheckDivisibility(n, 5) {
			fmt.Println("Buzz")
		} else {
			fmt.Println(n)
		}
	}
}

func CheckDivisibility(dividend int, divisor int) bool {
	return dividend % divisor == 0
}

File: FizzBuzz.rb
#!/usr/bin/ruby
def check_divisibility(dividend, divisor)
    return dividend % divisor == 0
end
def fizz_buzz(n, max)
  until n == max
    n += 1
    if check_divisibility(n,15)
      puts "FizzBuzz"
    elsif check_divisibility(n,3)
      puts "Fizz"
    elsif check_divisibility(n, 5)
      puts "Buzz"
    else
      puts n
    end
  end
end
fizz_buzz 0, 100


